scripts/version.sh

```diff
@@ -26,7 +26,7 @@ BASEDIR=$(dirname "$0")
 
 cd $BASEDIR/../
 
-if GIT_VERSION=$(git describe --tags --dirty --abbrev=7); then
+if GIT_VERSION=$(git describe --tags --dirty --abbrev=7 --match 'syslog-ng-*'); then
```

Comment:
I still would not include filtering based on tag name, as it should not care much about tag naming; I do not see any benefits via excluding tag names.
My assumption are the following: you must never tag master other then released stuff, otherwise it became very fast very ugly (seen git tag for every CI execution...)
If you tag a commit, you really want it to called by its name, so getting back that name like prefer to see: `best-feature-ever-1-g21ced7a` instead of `3.51-12313-g21ced7a`.

Reply:
This is a conceptual thing, so I don't mind accepting all tags, if you prefer to do so,  but in this case I sense some contradiction:

If we accept all tags as a version, why would we cut the `syslog-ng-` or the `v` prefix off from `syslog-ng-3.21.1`/`v3.21.1`? In my understanding, we do this because we have a specific expectation on how the version should look like.

Reply:
Nor I agree with `syslog-ng-` prefixed tags, nor I have all the knowledge why such tagging strategy is used; should just `3.20.1` be just fine, and there were no need to removing strings etc...

I would have been fine with `syslog-ng-3.20.1` format; but did the transformation in order not to break the builds created from released commits.

The `tr '-' '.'` is truly unfortunate thing :(, maybe I should have solved it in a different way, as packaging leaked into this version script.